compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirGetClassCallChecker.kt

```diff
@@ -73,6 +76,12 @@ object FirGetClassCallChecker : FirBasicExpressionChecker() {
         }
     }
 
+    private fun ConeKotlinType.isNullableTypeParameter(session: FirSession): Boolean {
+        if (this !is ConeTypeParameterType) return false
+        val typeParameter = lookupTag.typeParameterSymbol.fir
+        return !typeParameter.isReified && typeParameter.bounds.any { !it.coneType.isSubtypeOfAny(session) }
```

Comment:
What about the following case:
`fun <T> f2(t: T): Any where T : Any?, T : Comparable<T> = t::class`

This is ok in FE1.0 but would report the error with this change, since `Any?` is not a subtype of `Any`. I think this should be either `bounds.all` or get the intersection type.

Reply:
I forgot such upper bounds are intersection, thanks! We can simply rely on subtype check, simple, clearer! Also, updated a test.